### PR TITLE
Coffeescript -> ES6

### PR DESCRIPTION
**Pull request by lbntly**
+ rewrite w/ es6 syntax & new react stuff
+ remove coffeescript build stuff, add babel es6/react build stuff
+ add options to examples page
+ add eslint

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Oh nice! This is looking great.

I normally use the src for example pages as the src will be built by whatever is building the example page. This way you can iterate on both the src and examples w/o needing extra watchers/builders e.g. what I did for react-gravatar https://github.com/KyleAMathews/react-gravatar/blob/master/www/pages/index.js

Actually on that subject — this doesn't need to be part of this PR but it would be nice to convert the examples page to use Gatsby as has everything needed for docs page built in (hot dev, build step, deploy script for github pages, etc.)

Have you setup eslint before? It'd be nice to get that in as well. I normally use Airbnb's config.

Lastly I tried running this and it looks like `babel-cli` didn't make it into the `devDependencies`.

Thanks for making this happen!

**Comment by lbntly** [comment]
@KyleAMathews Great points -- I probably won't be adding Gatsby in this pr but I think adding eslint would help a lot w/ polish. I'll have an update soon!

File: package.json [hunk 1]
```diff
@@ -13,18 +13,26 @@
   },
   "dependencies": {
     "classnames": "^2.2.3",
-    "object-assign": "^4.1.0"
+    "object-assign": "^4.1.0",
+    "react": "^15.3.0",
```
Comment (KyleAMathews) [inline package.json]: `react` + `react-dom` should actually be devDependencies. NPM is sorta to blame for this. The problem that can arise is that if when say, React 16 is released, a user installs this module, NPM will then install both React 16 & 15 and will add both to the bundle.js. If you leave them off the dependencies then react-spinkit will just require the react that's already being used. Moving these to the devDependencies is still needed for the examples.

Comment (lbntly) [inline package.json]: oops, yeah, I moved them here to make eslint shut up but meant to put them in `peerDependencies` :)

**Comment by KyleAMathews** [comment]
Published 1.1.9!!! Thanks again! Awesome work. Also we have a shiny-ier new site courtesy of Gatsby http://kyleamathews.github.io/react-spinkit/

One other quick change I made before publishing you might be interested in — I removed the peerDependencies. It's kind of a crappy API. NPM 2 treats them very differently than NPM 3 and they tend to be far more confusing than helpful. The only time I'd say for sure use peerDependencies is if your component depends on a specific version of say React e.g. it doesn't work before React 0.14. Then peerDependencies are helpful as they ensure a version dependency. But since react-spinkit works for almost any version of React, peerDependencies aren't helpful.

**Pull request merged**